File: DuDoong-Api/src/main/java/band/gosrock/api/email/service/HostUserInvitationEmailService.java
package band.gosrock.api.email.service;


import band.gosrock.domain.domains.host.domain.HostRole;
import band.gosrock.infrastructure.config.mail.dto.EmailUserInfo;
import band.gosrock.infrastructure.config.ses.AwsSesUtils;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.thymeleaf.context.Context;

@Service
@RequiredArgsConstructor
public class HostUserInvitationEmailService {
    private final AwsSesUtils awsSesUtils;

    public void execute(EmailUserInfo userInfo, String hostName, HostRole hostRole) {
        Context context = new Context();
        context.setVariable("userInfo", userInfo);
        context.setVariable("hostName", hostName);
        context.setVariable("role", hostRole.getValue());
        awsSesUtils.singleEmailRequest(
                userInfo.getEmail(), "두둥" + hostName + " 호스트 초대 알림 드립니다.", "hostInvite", context);
    }
}

File: DuDoong-Api/src/main/kotlin/band/gosrock/api/email/service/HostUserInvitationEmailService.kt
package band.gosrock.api.email.service

import band.gosrock.domain.domains.host.domain.HostRole
import band.gosrock.infrastructure.config.mail.dto.EmailUserInfo
import band.gosrock.infrastructure.config.ses.AwsSesUtils
import org.springframework.stereotype.Service
import org.thymeleaf.context.Context

@Service
class HostUserInvitationEmailService(
    private val awsSesUtils: AwsSesUtils,
) {
    fun execute(userInfo: EmailUserInfo, hostName: String, hostRole: HostRole) {
        val context = Context()
        context.setVariable("userInfo", userInfo)
        context.setVariable("hostName", hostName)
        context.setVariable("role", hostRole.value)
        awsSesUtils.singleEmailRequest(
            userInfo,
            "두둥$hostName 호스트 초대 알림 드립니다.",
            "hostInvite",
            context,
        )
    }
}
